### PR TITLE
Allow setting custom replacement templates

### PR DESCRIPTION
**Pull request by speshak**
This allows adjusting where in the message the ticket number is added.

### CODE REVIEW

**Comment by milin** [comment]
Thanks for the pr. The custom template will be useful. Can you take a look at why the tests are failing?

**Comment by speshak** [comment]
Yeah, I'll give those failures a look.

File: giticket/giticket.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def update_commit_message(filename, regex):
             re.search(regex, branch),
         ]):
             ticket = branch.split(six.text_type('_'))[0]
-            new_commit_msg = '{} {}'.format(ticket, commit_msg)
+            new_commit_msg = format_string.format(ticket=ticket, commit_msg=commit_msg)
```
Comment (milin) [inline giticket/giticket.py:25]: Can you update to 
```
new_commit_msg = format_string.format(ticket=ticket, commit_msg=commit_msg.strip())
```

Currently `commit-msg` has a newline character which makes the commit message and ticket be in two separate lines, when --format is passed in.

```
Date:   Fri Jan 25 11:18:02 2019 -0500

    test 
     JIRA-7633
```

**Review by milin — CHANGES_REQUESTED** [review]
Minor change required

File: giticket/giticket.py (hunk 1 above)
Comment (speshak) [reply giticket/giticket.py:25]: Doh! My immediate use case was for simple single line commit messages.  I'm going to think about this a little bit and get a better solution put together.  My initial thought is to split `commit_msg` into `commit_subject` and `commit_body` tokens.  Your original format would be represented as:

    {ticket} {commit_subject}
    {commit_body}

And my intent would be:

    {commit_subject} {ticket}
    {commit_body}

**Comment by Rictus** [comment]
Hello
I'm also interested in this pull request, any update on when it will merge in ?

**Comment by milind-shakya-sp** [comment]
@Rictus looks like there was a slight minor issue. I can add a slight change on top of @speshak 's changes and we can have a custom template passed in. Be sure to update the hash.

**Comment by milind-shakya-sp** [comment]
@Rictus @speshak https://github.com/milin/giticket/commit/65ce90fb60fea9fb40c9da8fc045a86afdb76601 should now do the needful I think.

**Comment by milind-shakya-sp** [comment]
Closing in favor of https://github.com/milin/giticket/pull/5

**Comment by speshak** [comment]
👍 I forgot all about this PR.  Thanks for getting my rough idea cleaned up @Rictus & @milind-shakya-sp

**Pull request closed without merging**